src/main/java/com/altenheim/kalender/controller/logicController/SmartSearchController.java

```diff
@@ -0,0 +1,69 @@
+package com.altenheim.kalender.controller.logicController;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import com.altenheim.kalender.interfaces.ICalendarEntriesModel;
+import com.altenheim.kalender.interfaces.ISmartSearchController;
+import com.calendarfx.model.Entry;
+
+public class SmartSearchController implements ISmartSearchController 
+{
+	private ICalendarEntriesModel administrateEntries;
+
+	public SmartSearchController(ICalendarEntriesModel administrateEntries) 
+	{
+		this.administrateEntries = administrateEntries;
+	}
+	
+	public ArrayList<Entry<String>> findAvailableTimeSlot(Entry<String> input, int duration) 
+	{			
+		var result = administrateEntries.getSpecificCalendarByIndex(0).findEntries(
```

Comment:
Herz der Angelegenheit. Schönes Pairprogramming mit Nico S., diese Methode ermittelt freie Slots in Abhängigkeit des Nutzerwunsches (Parameter Entry) und speichert diese wiederum als Liste von neuen Entries. Dem wird noch eine Methode vorgelagert sein, die die Nutzereingaben aufbereitet (Anfahrt usw.) und eine nachgestellte Methode die dann in diesen Slots automatisch oder nach Userwunsch die Termine festlegt.